lib/make.js

```diff
@@ -15,11 +15,37 @@ function provideBuilder() {
     },
 
     settings: function (cwd) {
-      return [ {
+
+      var makefile = fs.realpathSync(path.join(cwd, 'Makefile'));
+
+      var createBuildConfig = function(name) {
+        return {
+          name: 'GNU Make: ' + name,
+          exec: 'make',
+          args: [ name ],
+          sh: false
+        };
+      };
+
+      var data = fs.readFileSync(makefile, 'utf8');
+      var lines = data.split(/\n/);
+      var config = [];
+
+      config.push({
         name: 'GNU Make: default',
         exec: 'make',
         sh: false
-      } ];
+      });
+
+      lines.forEach(function(line) {
+        var m = line.match(/^([\w\-]+):/);
```

Comment:
This is really tricky, which is why I've postponed this for now. Two issues are apparent with this solution:
  * This will match non-real such as `.PHONY`. While technically allowed to run those, it's generally discouraged.
  * This will not work with [included makefiles](https://www.gnu.org/software/make/manual/html_node/Include.html).